src/app/pages/reports-dashboard/reports.service.ts

```diff
@@ -35,47 +33,13 @@ export class ReportsService implements OnDestroy {
   private hasNfs = false;
   private hasPartitions = false;
 
+  private legendEventEmitter$ = new Subject<LegendDataWithStackedTotalHtml>();
+  readonly legendEventEmitterObs$ = this.legendEventEmitter$.asObservable();
```

Comment:
This is unnecessary as subjects are already observables.

Reply:
I only do this because subjects allow emitting new values if you expose them directly. This subject is not supposed to emit values outside of this service. But if we expose it publicly, that becomes possible.